packages/core/lib/Diagnostics.ts

```diff
@@ -0,0 +1,36 @@
+export enum PluginEnum {
+  listener = "Listener",
+  crossover = "Crossover",
+  sampler = "Sampler",
+  searchAlgorithm = "Search algorithm",
+  terminationTrigger = "Termination trigger",
+  userInterface = "User-interface",
```

Comment:
Should we not define these plugin types in the plugin manager as an enum? It would not be logical to change the list here when we add additional types

Reply:
Agree